custom/benin/management/commands/migrate_users_and_their_cases_to_new_rc_level.py

```diff
@@ -72,7 +72,7 @@ def handle(self, domain, **options):
             users = _find_rc_users_at_location(domain, village)
             for user in users:
                 log(f"Total number of users for village: {village.name} found is {len(users)}")
-                user_rc_number = user.user_data.get('rc_number')
+                user_rc_number = user.get_user_data(domain).get('rc_number')
```

Comment:
just confirming that does not punt the work done by `prime_user_data_caches`?

Reply:
It does not. Just that, I did not find all properties when accessing `user_data` directly and looking further saw below comment on the model:
`    user_data = DictProperty()      # use get_user_data object instead of accessing this directly
`

Reply:
so prime_user_data_caches is suppose to populate user_data on users in bulk so that it can then be accessed.
It seems its defeating that.

Reply:
(Looked more into this now.)
I think It does that [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/users/user_data.py#L266) by adding it to the key value dict (`_user_data_accessors`) of the user. 
So while accessing it though `get_user_data` it does not need to populate User Data again.

Also seems like `user_data` field on Couch model is kinda deprecated (I think) and now moved to `SQLUserData` hence the need to access it through the method.

Reply:
Cool, thanks @ajeety4